node/src/accountant/big_int_db_processor.rs

```diff
@@ -0,0 +1,1613 @@
+            ),
+        }
+    }
+}
+
+//to be able to display things that implement ToSql
+pub trait ExtendedParamsMarker: ToSql + Display {}
+
+macro_rules! impl_of_extended_params_marker{
+    ($($implementer: ty),+) => {
+        $(impl ExtendedParamsMarker for $implementer {})+
+    }
+}
+
+impl_of_extended_params_marker!(i64, &str, Wallet);
+
+#[derive(Default)]
+pub struct SQLParamsBuilder<'a> {
+    key_spec_opt: Option<(&'a str, &'a str, &'a dyn ExtendedParamsMarker)>,
+    wei_change_spec_opt: Option<WeiChange>,
+    other_params: Vec<(&'a str, &'a dyn ExtendedParamsMarker)>,
+}
+
+impl<'a> SQLParamsBuilder<'a> {
+    pub fn key(
+        mut self,
+        table_param_name: &'a str,
+        substitution_name: &'a str,
+        value: &'a dyn ExtendedParamsMarker,
+    ) -> Self {
+        self.key_spec_opt = Some((table_param_name, substitution_name, value));
+        self
+    }
+
+    pub fn wei_change(mut self, wei_change: WeiChange) -> Self {
+        self.wei_change_spec_opt = Some(wei_change);
+        self
+    }
+
+    pub fn other(mut self, params: Vec<(&'a str, &'a (dyn ExtendedParamsMarker + 'a))>) -> Self {
+        self.other_params = params;
+        self
+    }
+
+    pub fn build(self) -> SQLParams<'a> {
+        let key_spec = self
+            .key_spec_opt
+            .unwrap_or_else(|| panic!("SQLparams cannot miss the component of a key"));
+        let wei_change_spec = self
+            .wei_change_spec_opt
+            .unwrap_or_else(|| panic!("SQLparams cannot miss the component of Wei change"));
+        let (wei_change_names, split_bytes) = Self::expand_wei_params(wei_change_spec);
+        let params = once((key_spec.1, key_spec.2))
+            .chain(self.other_params.into_iter())
+            .collect();
+        SQLParams {
+            table_key_name: key_spec.0,
+            wei_change_params: [
+                (wei_change_names.0, split_bytes.0),
+                (wei_change_names.1, split_bytes.1),
+            ],
+            params_except_wei_change: params,
+        }
+    }
+
+    fn expand_wei_params(wei_change_spec: WeiChange) -> ((String, String), (i64, i64)) {
+        let (name, num): (&'static str, i128) = match wei_change_spec {
+            Addition(name, num) => (name, checked_conversion::<u128, i128>(num)),
+            Subtraction(name, num) => (name, checked_conversion::<u128, i128>(num).neg()),
+        };
+        let (high_bytes, low_bytes) = BigIntDivider::deconstruct(num);
+        let param_sub_name_for_high_bytes = Self::proper_wei_change_param_name(name, High);
+        let param_sub_name_for_low_bytes = Self::proper_wei_change_param_name(name, Low);
+        (
+            (param_sub_name_for_high_bytes, param_sub_name_for_low_bytes),
+            (high_bytes, low_bytes),
+        )
+    }
+
+    fn proper_wei_change_param_name(base_word: &str, byte_order: ByteOrder) -> String {
+        format!(":{}_{}_b", base_word, byte_order)
+    }
+}
+
+enum ByteOrder {
```

Comment:
"Byte order" already has an established meaning in the industry. The two values are little-endian and big-endian. It has to do with whether a multibyte number like 0x1234 will be stored with the 0x12 first and the 0x34 second (big-endian) or with the 0x34 first and the 0x12 second (little-endian).

This doesn't seem to have anything to do with that kind of byte order. I recommend a name change. Maybe...hmm...ValueHalf? QuadWordPosition? Yuck. Maybe you can come up with something better.